stac_fastapi/pgstac/stac_fastapi/pgstac/core.py

```diff
@@ -180,9 +181,15 @@ async def get_item(self, item_id: str, collection_id: str, **kwargs) -> Item:
         Returns:
             Item.
         """
+        # If collection does not exist, NotFoundError wil be raised
+        await self.get_collection(collection_id, **kwargs)
```

Comment:
Same question as above regarding the benefits of this safeguard

Reply:
Same idea as above - checking if a collection exists because if not a 404 will be returned